ppr-ui/src/App.vue

```diff
@@ -7,54 +7,52 @@
     feature-two
 </template>
 
-<script>
-    const APP_PATH = process.env.VUE_APP_PATH || 'app-path-foo-bar'
-    const DefaultLayout = 'public'
-    import AppData from '@/utils/app-data'
-    import FeatureOne from '@/components/FeatureOne'
-    import FeatureTwo from '@/components/FeatureTwo'
-
-    export default {
-        components: { FeatureOne, FeatureTwo },
-
-        data: function () {
-            return {
-                dataLoaded: false,
-            }
-        },
-        provide() {
-            return {
-                originUrl: this.origin,
-                authApiUrl: this.authApiUrl
-            }
-        },
-        computed: {
-            layout() {
-                return (this.$route.meta.layout || DefaultLayout) + '-layout'
-            },
-            origin() {
-                const root = window.location.origin || ''
-                const path = APP_PATH
-                return `${root}/${path}`
-            },
-            authAPIURL() {
-                return sessionStorage.getItem('AUTH_API_URL')
-            },
-        },
-        methods: {
-            setLoaded: function (flag) {
-                this.dataLoaded = flag
-            }
-        },
-        errorCaptured(err, vm, info) {
-            console.log('App errorCaptured', err)
-            // err: error trace
-            // vm: component in which error occured
-            // info: Vue specific error information such as lifecycle hooks, events etc.
-            // TODO: Perform any custom logic or log to server
-            // return false to stop the propagation of errors further to parent or global error handler
-        }
+<script lang="ts">
+  import {computed, createComponent, onErrorCaptured, provide, ref} from "@vue/composition-api";
+  import {Data} from "@vue/composition-api/dist/component";
+  import {provideRouter, useRouter} from "@/router/router";
+  import FeatureOne from '@/components/FeatureOne.vue'
+  import FeatureTwo from '@/components/FeatureTwo.vue'
+  import AppData from "@/utils/app-data";
+
+  const APP_PATH = process.env.VUE_APP_PATH || 'app-path-foo-bar'
+  const DefaultLayout = 'public'
+
+  function origin(): string {
+    const root = window.location.origin || ''
+    const path = APP_PATH
+    return `${root}/${path}`
+  }
+
+  function authAPIURL(): string {
+    return sessionStorage.getItem('AUTH_API_URL')
+  }
```

Comment:
In the previous version, these were exposed as computed, but they were not being used in the template so I removed them from the component.  They're just referenced in `setup()` now.